service/service_test.go

```diff
@@ -38,6 +41,10 @@ import (
 	"go.opentelemetry.io/collector/component/componenttest"
 	"go.opentelemetry.io/collector/config"
 	"go.opentelemetry.io/collector/config/configmodels"
+	"go.opentelemetry.io/collector/processor/attributesprocessor"
+	"go.opentelemetry.io/collector/processor/batchprocessor"
+	"go.opentelemetry.io/collector/receiver/jaegerreceiver"
+	"go.opentelemetry.io/collector/service/builder"
```

Comment:
I am not sure I like how we depend on specific components from here. Can we use `componenttest` and example components for testing instead?

I would prefer to minimize the coupling between this package and actual component packages.

Reply:
The test was already using `"go.opentelemetry.io/collector/service/defaultcomponents"`. The components were parsed but not accessed directly. For instance this config was used by this test https://github.com/open-telemetry/opentelemetry-collector/blob/9778b16b81016d6035b2a9f42cda8dc05d8165b8/service/testdata/otelcol-config.yaml#L1